sql/core/src/test/scala/org/apache/spark/sql/DateFunctionsSuite.scala

```diff
@@ -103,6 +153,51 @@ class DateFunctionsSuite extends QueryTest with SharedSQLContext {
       Row("2015", "2015", "2013"))
   }
 
+  test("date format with session local timezone") {
+    val df = Seq((d, sdf.format(d), ts)).toDF("a", "b", "c")
+
+    // The child of date_format is implicitly casted to TimestampType with session local timezone.
+    //
+    // +---+---------------------+-------------+---------------------+
+    // |   | df                  | timestamp   | date_format         |
+    // +---+---------------------+-------------+---------------------+
+    // | a |                16533|1428476400000|"2015-04-08 00:00:00"|
+    // | b |"2015-04-08 13:10:15"|1428523815000|"2015-04-08 13:10:15"|
```

Comment:
I'm a little confused, the `d` is already a `Date`, how can we have the time info back after converting the date to string?